test/test_h_transport_problem.py

```diff
@@ -207,3 +207,240 @@ def test_update_time_dependent_values_temperature(T_function, expected_values):
             computed_value = my_model.temperature_fenics.vector.array[-1]
             print(computed_value)
         assert np.isclose(computed_value, expected_values[i])
+
+
+def test_initialise_exports_find_species_with_one_field():
+    """Test that a species can be found from the model species if given as a string"""
+
+    # BUILD
+    my_model = F.HydrogenTransportProblem(
+        mesh=F.Mesh1D(vertices=np.array([0.0, 1.0, 2.0, 3.0, 4.0])),
+        species=[F.Species("H"), F.Species("D")],
+        temperature=1,
+    )
+    surf = F.SurfaceSubdomain1D(id=1, x=1)
+
+    my_model.exports = [F.SurfaceFlux(field="J", surface=surf)]
+    my_model.define_function_spaces()
+
+    # TEST
+    with pytest.raises(ValueError, match="Species J not found in list of species"):
+        my_model.initialise_exports()
+
+
+def test_define_D_global_different_temperatures():
+    """Test that the D_global attribute is correctly defined when the temperature
+    is different in the volume subdomains"""
+    D_0, E_D = 1.5, 0.1
+    my_mat = F.Material(D_0=D_0, E_D=E_D, name="my_mat")
+    surf = F.SurfaceSubdomain1D(id=1, x=0)
+    H = F.Species("H")
+
+    my_model = F.HydrogenTransportProblem(
+        mesh=F.Mesh1D(np.linspace(0, 4, num=101)),
+        subdomains=[
+            F.VolumeSubdomain1D(id=1, borders=[0, 2], material=my_mat),
+            F.VolumeSubdomain1D(id=2, borders=[2, 4], material=my_mat),
+        ],
+        species=[H],
+        temperature=lambda x: 100.0 * x[0] + 50,
+        exports=[
+            F.SurfaceFlux(
+                field=H,
+                surface=surf,
+            ),
+        ],
+    )
+
+    my_model.define_function_spaces()
+    my_model.define_markers_and_measures()
+    my_model.define_temperature()
+
+    D_computed, D_expr = my_model.define_D_global(H)
+
+    computed_values = [D_computed.x.array[0], D_computed.x.array[-1]]
+
+    D_analytical_left = D_0 * np.exp(-E_D / (F.k_B * 50))
+    D_analytical_right = D_0 * np.exp(-E_D / (F.k_B * 450))
+
+    expected_values = [D_analytical_left, D_analytical_right]
+
+    assert np.isclose(computed_values, expected_values).all()
+
+
+def test_define_D_global_different_materials():
+    """Test that the D_global attribute is correctly defined when the material
+    is different in the volume subdomains"""
+    D_0_left, E_D_left = 1.0, 0.1
+    D_0_right, E_D_right = 2.0, 0.2
+    my_mat_L = F.Material(D_0=D_0_left, E_D=E_D_left, name="my_mat_L")
+    my_mat_R = F.Material(D_0=D_0_right, E_D=E_D_right, name="my_mat_R")
+    surf = F.SurfaceSubdomain1D(id=1, x=0)
+    H = F.Species("H")
+
+    my_model = F.HydrogenTransportProblem(
+        mesh=F.Mesh1D(np.linspace(0, 4, num=101)),
+        subdomains=[
+            F.VolumeSubdomain1D(id=1, borders=[0, 2], material=my_mat_L),
+            F.VolumeSubdomain1D(id=2, borders=[2, 4], material=my_mat_R),
+        ],
+        species=[F.Species("H"), F.Species("D")],
```

Comment:
1) Why two species?
2) You seem to define twice F.Species("H"), here and on line 281 just above